service/api_logic/managers/recommendation_menager.py

```diff
@@ -1,22 +1,28 @@
 import pandas as pd
 from database.azure_blob_storage.save_get_blob import blob_get_news, blob_get_news_bulk
+from database.postgres.dal.user import UserDAL
+from exept.exeptions import UserDoesNotExistError
 from service.api_logic.helpers.calculating_helper import CalculatingRecommendationHelper
 from service.api_logic.news_logic import NewsService
 from service.api_logic.user_logic import UserService
 
 
 class RecommendationManager:
-    def __init__(self, user_service: UserService, news_service: NewsService):
+    def __init__(self, user_service: UserService, news_service: NewsService, user_dal: UserDAL):
         self.user_service = user_service
         self.news_service = news_service
+        self.user_dal = user_dal
         self.calculating_helper = CalculatingRecommendationHelper()
 
 
-    def get_recommended_news_for_user(self, user_id: int):
-        preferred_teams, preferred_sports = self.user_service.get_user_sport_and_club_preferences(user_id)
+    def get_recommended_news_for_user(self, user_email: str):
+        user_id = self.user_dal.get_user_id_by_email(user_email)
+        if not user_id:
+            raise UserDoesNotExistError(user_id)
+        preferred_teams, preferred_sports = self.user_service.get_user_sport_and_club_preferences(user_id[0])
 
         user_unseen_news_df, recs_by_user_preferences, recs_by_user_last_watch = self.news_service.user_recommendations_based_on_preferences_and_last_watch(
-            user_id,
+            user_id[0],
```

Comment:
lines 25 and 35 are the same. Maybe we should first extract user_id[o] into some varuiable?